File: GRAFOS/2419/main.js
function coastSize(m, n) {
    let size = 0;
    let map = _getMap(m);
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            if (_isCoast(map, i, j, m, n)) {
                size++;
            }
        }
    }
    return size;
}

function _getMap(m) {
    map = [];
    for (let i = 1; i <= m; i++) {
        map.push(lines[i]);
    }
    return map;
}

function _isCoast(map, i, j, m, n) {
    if (map[i][j] == '#') {
        if (i === 0 || i === m - 1 || j === 0 || j === n - 1) {
            return true;
        }
        if (map[i - 1][j] == '.' || map[i + 1][j] == '.' || map[i][j - 1] == '.' || map[i][j + 1] == '.') {
            return true;
        }
    }
    return false;
}

let input = require('fs').readFileSync('/dev/stdin', 'utf-8');
var lines = input.split('\n');
let [m, n] = lines[0].split(' ').map(Number);
console.log(coastSize(m, n));

File: GRAFOS/2419/main.py
def coast_size(m, n):
    size = 0
    map = _get_map(m)
    for i in range(m):
        for j in range(n):
            if _is_coast(map, i, j, m, n):
                size += 1
    return size


def _get_map(m):
    map = []
    for _ in range(m):
        map.append(input())
    return map


def _is_coast(map, i, j, m, n):
    if map[i][j] == '#':
        if i == 0 or i == m-1 or j == 0 or j == n-1:
            return True
        if map[i-1][j] == '.' or map[i+1][j] == '.' or map[i][j-1] == '.' or map[i][j+1] == '.':
            return True
    return False


m, n = map(int, input().split())
print(coast_size(m, n))
